worlds/yoshisisland/Rules.py

```diff
@@ -0,0 +1,597 @@
+    set_rule(multiworld.get_location("Make Eggs, Throw Eggs: Level Clear", player), lambda state: state.has_all({'Beanstalk'}, player))
+
+    set_rule(multiworld.get_location("Watch Out Below!: Red Coins", player), lambda state: state.has_all({'Large Spring Ball', 'Helicopter Morph'}, player))
+    set_rule(multiworld.get_location("Watch Out Below!: Flowers", player), lambda state: state.has_all({'Large Spring Ball', 'Helicopter Morph'}, player))
+    set_rule(multiworld.get_location("Watch Out Below!: Stars", player), lambda state: state.has_all({'Large Spring Ball'}, player) and logic.has_midring(state))
+    set_rule(multiworld.get_location("Watch Out Below!: Level Clear", player), lambda state: state.has_all({'Large Spring Ball'}, player))
+
+    set_rule(multiworld.get_location("The Cave Of Chomp Rock: Red Coins", player), lambda state: state.has_all({'Chomp Rock'}, player))
+    set_rule(multiworld.get_location("The Cave Of Chomp Rock: Flowers", player), lambda state: state.has_all({'Chomp Rock'}, player))
+
+    set_rule(multiworld.get_location("Burt The Bashful's Fort: Red Coins", player), lambda state: state.has_all({'Spring Ball'}, player))
+    set_rule(multiworld.get_location("Burt The Bashful's Fort: Flowers", player), lambda state: state.has_all({'Spring Ball', 'Key'}, player) and (state.has('Egg Capacity Upgrade', player, 3) or logic.combat_item(state)))
+    set_rule(multiworld.get_location("Burt The Bashful's Fort: Stars", player), lambda state: state.has_all({'Spring Ball'}, player) and (logic.has_midring(state) or state.has('Key', player)))
+
+    set_rule(multiworld.get_location("Hop! Hop! Donut Lifts: Stars", player), lambda state: logic.has_midring(state) or logic.cansee_clouds(state))
+
+    set_rule(multiworld.get_location("Shy-Guys On Stilts: Red Coins", player), lambda state: state.has_all({'Large Spring Ball', 'Flashing Eggs', 'Mole Tank Morph', '! Switch'}, player))
+    set_rule(multiworld.get_location("Shy-Guys On Stilts: Flowers", player), lambda state: state.has_all({'Large Spring Ball'}, player))
+    set_rule(multiworld.get_location("Shy-Guys On Stilts: Stars", player), lambda state: (logic.has_midring(state) and state.has('Tulip', player) or logic.has_midring(state) and state.has('Beanstalk', player)) and state.has('Large Spring Ball', player))
+    set_rule(multiworld.get_location("Shy-Guys On Stilts: Level Clear", player), lambda state: state.has_all({'Large Spring Ball', 'Beanstalk'}, player))
+
+    set_rule(multiworld.get_location("Touch Fuzzy Get Dizzy: Red Coins", player), lambda state: state.has_all({'Flashing Eggs', 'Spring Ball', 'Chomp Rock', 'Beanstalk'}, player))
+    set_rule(multiworld.get_location("Touch Fuzzy Get Dizzy: Stars", player), lambda state: lambda state: logic.has_midring(state) or (logic.cansee_clouds and state.has_all({'Spring Ball', 'Chomp Rock', 'Beanstalk'}, player)))

```

Comment:
```suggestion
    set_rule(multiworld.get_location("Touch Fuzzy Get Dizzy: Stars", player), lambda state: logic.has_midring(state) or (logic.cansee_clouds and state.has_all({'Spring Ball', 'Chomp Rock', 'Beanstalk'}, player)))

```

Reply:
How did this one even happen without anyone seeing it until now lmao